test/pipe.js

```diff
@@ -76,8 +107,6 @@ if (process.platform !== 'win32') {
     });
   } else {
     console.error('Warning: Cannot verify piped exec');
+    shell.exit(123);
```

Comment:
I’m going to guess that no one verified that the tests work on unix when grep is not available because, with grep being part of the POSIX spec, it’s hard to find such a unix system. This whole conditional on `shell.which('grep').stdout` should probably just be removed because you can assume grep exists when `process.platform !== 'win32'`.